packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -258,6 +258,9 @@ class FlutterDevice {
             handleError(e);
             return;
           } else {
+            if (device.dds.uri == null) {
+              rethrow;
```

Comment:
Should this be `handleError(e); return;`?

Reply:
`handleError` won't actually surface the DDS exception, just notify that we've failed to connect to the VM service. However, completing the completer with the original exception is probably the right thing to do.

Reply:
Huh? Then I'm confused. `handleError()` is also completing the completer with an error and has the additional benefit of already printing helpful log messages, and checking whether the completer is already completed.

Reply:
`handleError()` will swallow some information (the original exception, and its stack trace) and, in its current form, will only raise an exception stating 'failed to connect to http://localhost:1234`, making it harder to find the actual issue in the crash reporter when it does happen. I can change it if you'd like, but that's what my thought process was.